src/ZODB/utils.py

```diff
@@ -84,13 +84,23 @@ def byte_chr(int):
 # The distinction between ints and longs is blurred in Python 2.2,
 # so u64() are U64() really the same.
 
+_OID_PACK = struct.Struct(">Q").pack
+_OID_UNPACK = _OID_PACK.__self__.unpack
+
+
 def p64(v):
-    """Pack an integer or long into a 8-byte string"""
-    return pack(">Q", v)
+    """Pack an integer or long into a 8-byte string."""
+    try:
+        return _OID_PACK(v)
+    except struct.error as e:
+        raise ValueError(*(e.args + (v,)))
```

Comment:
It does, yes. On Python 3, there's no more '.message' attribute, so it needs to. And that's why the tests are only checking args[-1], because the message string in args[0] varies a lot.